service-base/src/main/java/org/eclipse/hono/service/auth/device/UsernamePasswordCredentials.java

```diff
@@ -114,27 +115,31 @@ public final String getPassword() {
     @Override
     public boolean matchesCredentials(final JsonObject candidateSecret) {
 
+        encoder = PasswordEncoderFactory.createDelegatingPasswordEncoder();
```

Comment:
I would hope that we can re-use the same instance for all invocations, i.e. that the `PasswordEncoders` are thread safe. Can you verify?

Reply:
I found nothing that indicate this. This credentials objects are not reusable and encoders are not heavy-stated, so I think it's not too big of an issue. We can add that to the list of future improvements in this area.